File: Tareas/Tarea-4/src/models/client.model.js
import mongoose from 'mongoose';

const clientSchema = new mongoose.Schema({
    ci: { type: String, required: true, unique: true, trim: true, maxlength: 20 },
    firstName: { type: String, required: true, trim: true, maxlength: 100 },
    lastName: { type: String, required: true, trim: true, maxlength: 100 },
    sex: { type: String, enum: ['M', 'F', 'O'], required: true }
}, { timestamps: true });

export default mongoose.model('Client', clientSchema);

File: Python/ApiREST - Tarea 4/app/models/client_model.py
import mongoengine as me
from datetime import datetime

class Client(me.Document):
    ci = me.StringField(required=True, unique=True, max_length=20)
    firstName = me.StringField(required=True, max_length=100)
    lastName  = me.StringField(required=True, max_length=100)
    sex       = me.StringField(required=True, choices=("M", "F", "O"))

    # Mapeo a createdAt/updatedAt
    created_at = me.DateTimeField(db_field="createdAt", default=datetime.utcnow)
    updated_at = me.DateTimeField(db_field="updatedAt", default=datetime.utcnow)
    version    = me.IntField(db_field="__v")

    meta = {
        "collection": "clients",
        "indexes": [
            {"fields": ["ci"], "unique": True},
            {"fields": ["lastName", "firstName"]},
        ],
    }

    def save(self, *args, **kwargs):
        self.updated_at = datetime.utcnow()
        return super().save(*args, **kwargs)
